src/psyclone/tests/parse/algorithm_test.py

```diff
@@ -0,0 +1,295 @@
+        in str(excinfo.value)
+
+
+def test_parsefp2_invalid_fortran(tmpdir):
+    '''Test that if the Fortran contained in the file specified in the
+    filename argument then an exception is raised in the expected
+    way. *** Create the parse_tree in-place rather than running
+    PSyclone. Once created make the parse_tree content invalid using
+    monkeypatch.
+
+    '''
+    my_file = str(tmpdir.join("invalid.f90"))
+    ffile = open(my_file, "w")
+    ffile.write("invalid Fortran code")
+    ffile.close()
+    with pytest.raises(ParseError) as excinfo:
+        _ = parse_fp2(my_file)
+    assert "Syntax error in file" in str(excinfo.value)
+
+
+# function get_invoke_label() tests
+
+
+def test_getinvokelabel_invalid_tree():
+    '''Test that if the parse tree argument is not an Actual_Arg_Spec
+    then an exception is raised in the expected
+    way.
+
+    '''
+    with pytest.raises(InternalError) as excinfo:
+        _ = get_invoke_label("invalid", "dummy.f90")
+    assert (
+        "Expected a Fortran argument of the form name=xxx but found instance "
+        "of") in str(excinfo.value)
+
+
+def test_getinvokelabel_invalid_items(monkeypatch):
+    '''Test that if the parse tree argument is an Actual_Arg_Spec but does
+    not contain two items then an exception is raised in the expected
+    way. Create the parse_tree in-place rather than running
+    PSyclone. Once created make the parse_tree content invalid using
+    monkeypatch.
+
+    '''
+    from fparser.two.Fortran2003 import Actual_Arg_Spec
+    from fparser.two.parser import ParserFactory
+    _ = ParserFactory().create(std="f2003")
+    parse_tree = Actual_Arg_Spec("name='myname'")
+    monkeypatch.setattr(parse_tree, "items", [None, None, None])
+    with pytest.raises(InternalError) as excinfo:
+        _ = get_invoke_label(parse_tree, "dummy.f90")
+    assert (
+        "Expected the Fortran argument to have two items but found "
+        "'3'.") in str(excinfo.value)
+
+# function get_kernel() tests
+
+
+def test_getkernel_invalid_tree():
+    '''Test that if the get_kernel function is passed an invalid
+    parse tree argument, then it raises an exception
+    in the expected way.
+
+    '''
+    with pytest.raises(InternalError) as excinfo:
+        _ = get_kernel("invalid", "dummy.f90")
+    assert (
+        "Expected a parse tree (type Part_Ref) but found instance of ") \
+        in str(excinfo.value)
+
+
+def test_getkernel_invalid_children(monkeypatch):
+    '''Test that if the get_kernel function finds Part_Ref as the top
+    level of the parse tree but this does not have two children then
+    it raises an exception in the expected way. Create the parse_tree
+    in-place rather than running PSyclone. Once created make the
+    parse_tree content invalid using monkeypatch.
+
+    '''
+    from fparser.two.Fortran2003 import Part_Ref
+    from fparser.two.parser import ParserFactory
+    _ = ParserFactory().create(std="f2003")
```

Comment:
Fixture please.

Reply:
Replaced all cases of ParserFactory with a fixture in this file.